### PR TITLE
Sort data by upload date and creation date

### PR DESCRIPTION
**Pull request by lvreynoso**
### Description

Sorts trees and samples by upload / creation date.

#### Issue
[ch137016](https://app.clubhouse.io/genepi/story/137016)

### Test plan

Write how your changes are tested, or give a convincing reason why they can't be tested automatically.

### CODE REVIEW

File: src/frontend/src/views/Data/index.tsx [hunk 1]
```diff
@@ -78,6 +83,20 @@ const Data: FunctionComponent = () => {
     category.data = transformedData;
   });
 
+  // sort data by creation date
+  dataCategories.forEach((category) => {
+    if (category.data === undefined) {
+      return;
+    }
+    const sortKey = sortByKeys[category.text]
+    console.log(sortKey)
```
Comment (tihuan) [inline src/frontend/src/views/Data/index.tsx]: nit: unneeded console.log 😄

File: src/frontend/src/views/Data/index.tsx [hunk 2]
```diff
@@ -78,6 +83,20 @@ const Data: FunctionComponent = () => {
     category.data = transformedData;
   });
 
+  // sort data by creation date
+  dataCategories.forEach((category) => {
+    if (category.data === undefined) {
```
Comment (tihuan) [inline src/frontend/src/views/Data/index.tsx:88]: nit: `!category.data` would do instead of `category.data === undefined`

**Review by tihuan — APPROVED** [review]
LGTM after two small nits! Thanks @lvreynoso 🎉 !

File: src/frontend/src/views/Data/index.tsx (hunk 2 above)
Comment (lvreynoso) [reply src/frontend/src/views/Data/index.tsx:88]: Typescript needs you to explicitly rule out undefined in order for type checking to work though.

Comment (tihuan) [reply src/frontend/src/views/Data/index.tsx:88]: ah right, `if (!category.data)` should have the same effect as `if (category.data === undefined)`, just less code? Or does TS do something different in those two cases? 

Thanks!

**Pull request merged**